btf/btf.go

```diff
@@ -199,26 +199,39 @@ func loadSpecFromELF(file *internal.SafeELFFile) (*Spec, error) {
 		return nil, fmt.Errorf("compressed BTF is not supported")
 	}
 
-	return loadRawSpec(btfSection.ReaderAt, file.ByteOrder, sectionSizes, vars)
+	return loadRawSpec(btfSection.ReaderAt, file.ByteOrder, nil, nil, sectionSizes, vars)
 }
 
-func loadRawSpec(btf io.ReaderAt, bo binary.ByteOrder, sectionSizes map[string]uint32, variableOffsets map[variable]uint32) (*Spec, error) {
```

Comment:
`loadRawSpec` is already too messy, I think this breaks the camel's back. I'll try to split it out some to make your life easier.

Reply:
See #708